docs/user/customization.md

```diff
@@ -64,4 +64,106 @@ An example `worker.ign` is shown below. It has been modified to increase the HTT
 }
 ```
 
+## Install time customization for machine configuration
+
+In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. daemonset, deployment, etc). For example, if a user wants to run a logging agent on all hosts, in general we would want to steer them towards deploying it as a daemonset object. However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster. A specific example that was given from a customer call was that due to a compliance mandate "they must run auditing the second the operating system comes up". In this case, the admin would need the ability to define a custom section of Ignition config (containing the systemd unit that runs the auditing container).
```

Comment:
"in general we would want to steer them towards deploying it as" -> "they will generally want"?  "steer" sounds somewhat manipulative, where what we're trying to do is point users at the simplest reliable workflow.

Reply:
> A specific example that was given from a customer call was that due to a compliance mandate "they must run auditing the second the operating system comes up".  In this case...

->

> For example, if you have a compliance mandate to start auditing "the second the operating system comes up", you may need to define a custom section of the Ignition configuration to enable an auditing systemd unit.

Or something?